### PR TITLE
uninstall.sh: add variable `NONINTERACTIVE` for non-interactive mode

### LINKED ISSUE #654: Suggestion: Support `NONINTERACTIVE=1` in the uninstall script
- [x] your problem was from running the official `install` or `uninstall` script?
- [x] after installation: ran `brew config` and `brew doctor` and included their output with your issue? If you couldn't install: provided your OS version with the output of your issue?

```
$ uname -a
Linux 3fc3e70d9324 5.4.0-109-generic #123-Ubuntu SMP Fri Apr 8 09:10:54 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

## What you were trying to do (and why)

Uninstall homebrew on a linux machine, non interactively, using the `uninstall` script:

```
NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
```

## What happened (include command output)

The command stopped to ask me if I am sure.

<details>
  <summary>Command output</summary>
  <pre>

Warning: This script will remove:
/home/linuxbrew/.linuxbrew/Caskroom/
/home/linuxbrew/.linuxbrew/Cellar/
/home/linuxbrew/.linuxbrew/Homebrew/
/home/linuxbrew/.linuxbrew/Homebrew/.dockerignore
/home/linuxbrew/.linuxbrew/Homebrew/.editorconfig
/home/linuxbrew/.linuxbrew/Homebrew/.github/
/home/linuxbrew/.linuxbrew/Homebrew/.gitignore
/home/linuxbrew/.linuxbrew/Homebrew/.shellcheckrc
/home/linuxbrew/.linuxbrew/Homebrew/.sublime/
/home/linuxbrew/.linuxbrew/Homebrew/.vale.ini
/home/linuxbrew/.linuxbrew/Homebrew/.vscode/
/home/linuxbrew/.linuxbrew/Homebrew/CHANGELOG.md
/home/linuxbrew/.linuxbrew/Homebrew/CONTRIBUTING.md
/home/linuxbrew/.linuxbrew/Homebrew/Dockerfile
/home/linuxbrew/.linuxbrew/Homebrew/LICENSE.txt
/home/linuxbrew/.linuxbrew/Homebrew/Library//
/home/linuxbrew/.linuxbrew/Homebrew/README.md
/home/linuxbrew/.linuxbrew/Homebrew/bin/brew
/home/linuxbrew/.linuxbrew/Homebrew/completions/
/home/linuxbrew/.linuxbrew/Homebrew/docs/
/home/linuxbrew/.linuxbrew/Homebrew/manpages/
/home/linuxbrew/.linuxbrew/bin/brew -> /home/linuxbrew/.linuxbrew/Homebrew/bin/brew
/home/linuxbrew/.linuxbrew/etc/bash_completion.d/brew -> /home/linuxbrew/.linuxbrew/Homebrew/completions/bash/brew
/home/linuxbrew/.linuxbrew/share/doc/homebrew -> /home/linuxbrew/.linuxbrew/Homebrew/docs
/home/linuxbrew/.linuxbrew/share/fish/vendor_completions.d/brew.fish -> /home/linuxbrew/.linuxbrew/Homebrew/completions/fish/brew.fish
/home/linuxbrew/.linuxbrew/share/man/man1/README.md -> /home/linuxbrew/.linuxbrew/Homebrew/manpages/README.md
/home/linuxbrew/.linuxbrew/share/man/man1/brew.1 -> /home/linuxbrew/.linuxbrew/Homebrew/manpages/brew.1
/home/linuxbrew/.linuxbrew/share/zsh/site-functions/_brew -> /home/linuxbrew/.linuxbrew/Homebrew/completions/zsh/_brew
/home/linuxbrew/.linuxbrew/var/homebrew/
/home/ubuntu/.cache/Homebrew/
Are you sure you want to uninstall Homebrew? This will remove your installed packages! [y/N]

  </pre>
</details>

## What you expected to happen

Expected the command to execute without stopping for user input, as if it was executed with `--force`.

## Step-by-step reproduction instructions (by running `brew` commands)

```bash
NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
```

is expected to behave the same as:

```bash
wget https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh
bash uninstall.sh --force
```

### PR DESCRIPTION
**Pull request by kshired**
Resolve #654 

Changes :
1. Add variable `NONINTERACTIVE` for non interactive mode. If this variable set to "1", this script will be executed as non interactive mode. In other case ( ex) NONINTERACTIVE is not set or not set to 1 ), it will be executed as interactive mode.

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @kshired!

File: uninstall.sh [hunk 1]
```diff
@@ -53,6 +53,7 @@ opt_force=""
 opt_quiet=""
 opt_dry_run=""
 opt_skip_cache_and_logs=""
+NONINTERACTIVE="${NONINTERACTIVE:-0}"
```
Comment (XuehaiPan) [inline uninstall.sh:56]:
```suggestion
```

**Comment by kshired** [comment]
> Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @kshired!

@MikeMcQuaid I changed single quote to double quote. Can u re-enable auto merge?

File: uninstall.sh [hunk 2]
```diff
@@ -200,14 +201,15 @@ homebrew_prefix_candidates=()
...
-Usage: $0 [options]
+Usage: [NONINTERACTIVE=1] $0 [options]
     -p, --path=PATH  Sets Homebrew prefix. Defaults to ${homebrew_prefix_default}.
         --skip-cache-and-logs
                      Skips removal of HOMEBREW_CACHE and HOMEBREW_LOGS.
     -f, --force      Uninstall without prompting.
     -q, --quiet      Suppress all output.
     -n, --dry-run    Simulate uninstall but don't remove anything.
     -h, --help       Display this message.
+    NONINTERACTIVE   Uninstall without prompting if NONINTERACTIVE is set to 1.
```
Comment (XuehaiPan) [inline uninstall.sh:212]:
```suggestion
    NONINTERACTIVE   Imply `--force` if NONINTERACTIVE is non-empty.
```

File: uninstall.sh [hunk 3]
```diff
@@ -342,7 +344,9 @@ then
   pretty_print_pathnames "${homebrew_files[@]}"
 fi
 
-if [[ -t 0 && -z "${opt_force}" && -z "${opt_dry_run}" ]]
+[[ "${NONINTERACTIVE}" -ne 1 ]] || ohai "Running in non-interactive mode because `$NONINTERACTIVE` is set."
+
+if [[ -t 0 && -z "${opt_force}" && -z "${opt_dry_run}" && "${NONINTERACTIVE}" -ne 1 ]]
```
[the remark this answers is not in the document]
Comment (XuehaiPan) [reply uninstall.sh:349]: We can set `opt_force` if `NONINTERACTIVE` is set.

**Comment by kshired** [comment]
@MikeMcQuaid @XuehaiPan 
I changed some codes based on suggestion. Thx.

**Comment by MikeMcQuaid** [comment]
Still some tabs that need to go to spaces I'm afraid: https://github.com/Homebrew/install/runs/6771657435?check_suite_focus=true#step:16:31

**Comment by kshired** [comment]
@MikeMcQuaid  I changed it. Thx

**Pull request merged**